### PR TITLE
Ensure correct error responses are returned when known exceptions occur

### LINKED ISSUE #4: RestfulHttpResponseToHttpResponseTransformer should not just blindly cast to RestfulHttpResponse
RestfulHttpResponseToHttpResponseTransformer should not just blindly cast to RestfulHttpResponse. There should be a check if the payload may be null. This happens in error cases and causes an incorrect error to propagate to the OpenHIM core.

We should probably construct a Http Response will the original error in this case.

### LINKED ISSUE #5: Ensure that correct error responses are returned when an exception occurs

### PR DESCRIPTION
**Pull request by rcrichton**
Fixes #5 
Fixes #4

### CODE REVIEW

**Comment by rcrichton** [comment]
@devcritter Could you please review this as well. Its a fix for RHIE as exception cases are not showing up correctly in the transaction log.

File: src/main/java/org/jembi/rhea/transformers/RestfulHttpResponseToHttpResponseTransformer.java [hunk 1]
```diff
@@ -14,12 +22,50 @@
...
+						msg.setPayload("Message orchestration failed, encounter enrichment failed: " + sw.toString());
+					} else if (cause instanceof LocationValidationException) {
+						msg.setProperty("http.status", 400, PropertyScope.OUTBOUND);
+						msg.setPayload("Message orchestration failed, the location ID failed validation: " + sw.toString());
+					} else if (cause instanceof ProviderValidationException) {
+						msg.setProperty("http.status", 400, PropertyScope.OUTBOUND);
+						msg.setPayload("Message orchestration failed, the provider ID failed validation: " + sw.toString());
+					} else {
+						msg.setProperty("http.status", 500, PropertyScope.OUTBOUND);
+						msg.setPayload("A server error has occured, the responce was not of type RestfulHttpResponse, caused by: " + sw.toString());
```
Comment (hnnesv) [inline src/main/java/org/jembi/rhea/transformers/RestfulHttpResponseToHttpResponseTransformer.java]: Spelling :) (response)

File: src/main/java/org/jembi/rhea/transformers/RestfulHttpResponseToHttpResponseTransformer.java [hunk 2]
```diff
@@ -14,12 +22,50 @@
...
+					}
+				}
+				
+			} else {
+				msg.setProperty("http.status", 500, PropertyScope.OUTBOUND);
+				msg.setPayload("A server error has occured, the responce was not of type RestfulHttpResponse");
```
Comment (hnnesv) [inline src/main/java/org/jembi/rhea/transformers/RestfulHttpResponseToHttpResponseTransformer.java]: Not a big deal or anything, but maybe you can add the payload class to the message (if it's not null) just to help debugging?

**Comment by hnnesv** [comment]
@rcrichton just added some super-minor comments, but it looks good :+1:

File: src/main/java/org/jembi/rhea/transformers/RestfulHttpResponseToHttpResponseTransformer.java (hunk 1 above)
Comment (rcrichton) [inline src/main/java/org/jembi/rhea/transformers/RestfulHttpResponseToHttpResponseTransformer.java]: Damn, I always make that mistake! :)

File: src/main/java/org/jembi/rhea/transformers/RestfulHttpResponseToHttpResponseTransformer.java (hunk 2 above)
Comment (rcrichton) [inline src/main/java/org/jembi/rhea/transformers/RestfulHttpResponseToHttpResponseTransformer.java]: Yeah, good idea. I will add it.

**Pull request merged**